core/src/main/java/hudson/model/queue/Executables.java

```diff
@@ -59,6 +71,49 @@
             }
         }
     }
+    
+    /**
+     * Get parent subtask from which the executable has been created.
+     * @param e Executable.
+     * @return Parent subtask from which the executable has been created.
+     *         {@code null} if the Executable is {@code null} OR has incompatible API (old plugin depending on a core older than 1.377)
+     * @since TODO
+     * @see #getParentOfOrFail(hudson.model.Queue.Executable) 
+     */
+    @CheckForNull
+    public static SubTask getParentOfOrNull(@CheckForNull Executable e) {
+        if (e == null) {
+            return null;
+        }
+        try {
+            return getParentOf(e);
+        } catch(RuntimeException | Error ex) {
+            LOGGER.log(Level.WARNING, formatUnsupportedExecutableAPIMessage(e), ex);
+            return null;
+        }
+    }
+    
+    /**
+     * Get parent subtask from which the executable has been created.
+     * @param e Executable
+     * @return Parent subtask from which the executable has been created
+     * @throws InvocationTargetException Operation failure due to the usage of incompatible API for old plugin depending on a core older than 1.377
```

Comment:
I do not think we need a new method here. Suffices to document the exception in the original, and have `doStop` let it be thrown up as it did before—right?